lib/rubocop/cop/style/initial_indentation.rb

```diff
@@ -9,28 +9,37 @@ module Style
       class InitialIndentation < Cop
         MSG = 'Indentation of first line in file detected.'.freeze
 
-        def investigate(processed_source)
-          first_token = processed_source.tokens.find do |t|
-            !t.text.start_with?('#')
-          end
-          return unless first_token
-          return if first_token.pos.column.zero?
-
-          with_space = range_with_surrounding_space(first_token.pos, :left,
-                                                    !:with_newline)
+        def investigate(_processed_source)
+          token = first_token
+          space_before(token) { |space| add_offense(space, token.pos) }
+        end
+
+        def autocorrect(range)
+          ->(corrector) { corrector.remove(range) }
+        end
+
+        private
+
+        def first_token
+          processed_source.tokens.find { |t| !t.text.start_with?('#') }
+        end
+
+        def space_before(token)
+          return unless token
+          return if token.pos.column.zero?
+
+          token_with_space =
+            range_with_surrounding_space(token.pos, :left, !:with_newline)
```

Comment:
Anything I haven't understood in a couple of seconds has been left untouched. :)